### PR TITLE
Set the zIndex higher for the tooltip where the mouse is. 

### LINKED ISSUE #16157: Graph: Primary tooltip drawn under shared tooltip
**What happened**:
Set the dashboard to use a shared tooltip.  When placing the cursor over one graph the tooltip from the graph below was drawn on top of the tooltip where the mouse was.

**What you expected to happen**:
The tooltip caused by the cursor should be on top.

**How to reproduce it (as minimally and precisely as possible)**:
Add two graphs with a number of series (10?) on top of each other.   Enable shared tooltip for the dashboard. Place the cursor close to the bottom of the top graph where both tooltips would be drawn.

**Anything else we need to know?**:

**Environment**:
- Grafana version:
6.0.2
- Data source type & version:
elasticsearch 6.4.2
- OS Grafana is installed on:
docker image
- User OS & Browser:
Windows 10, Chrome 72
- Grafana plugins:
Default
- Others:

Example:
OtherTooltipOnTop

### PR DESCRIPTION
**Pull request by cylon**
**What this PR does / why we need it**:
This keeps the "primary" tooltip on top of any overlapping tooltips.

**Which issue(s) this PR fixes**:
Fixes #16157 

**Special notes for your reviewer**:

**Release note**:

```release-note
The primary tooltip for the graph underneath the pointer is drawn on top of any other graph's tooltip.
```

### CODE REVIEW

File: public/app/plugins/panel/graph/graph_tooltip.ts [hunk 1]
```diff
@@ -49,11 +49,13 @@ export default function GraphTooltip(this: any, elem, dashboard, scope, getSerie
...
   };
 
-  this.renderAndShow = (absoluteTime, innerHtml, pos, xMode) => {
+  this.renderAndShow = (absoluteTime, innerHtml, pos, xMode, primary) => {
     if (xMode === 'time') {
       innerHtml = '<div class="graph-tooltip-time">' + absoluteTime + '</div>' + innerHtml;
     }
     $tooltip.html(innerHtml).place_tt(pos.pageX + 20, pos.pageY);
+    // The primary tooltip should be drawn on top
+    $tooltip.css('zIndex', primary ? 9999 : 9998);
```
Comment (torkelo) [inline public/app/plugins/panel/graph/graph_tooltip.ts]: what is not primary? can't see false ever being sent in

Comment (cylon) [reply public/app/plugins/panel/graph/graph_tooltip.ts]: It's due to the default value, like some of the other calls similar to the item argument, such as https://github.com/cylon/grafana/blob/master/public/app/plugins/panel/graph/graph.ts#L127.

File: public/sass/components/_tooltip.scss [hunk 2]
```diff
@@ -102,7 +102,6 @@
   font-weight: 200;
   background-color: $tooltipBackground;
   border-radius: 5px;
-  z-index: 9999;
```
Comment (torkelo) [inline public/sass/components/_tooltip.scss]: this could affect other tooltips, not sure

Comment (torkelo) [reply public/sass/components/_tooltip.scss]: yes this is used by other tooltips so have to remain here.

**Review by torkelo — CHANGES_REQUESTED** [review]

File: public/sass/components/_tooltip.scss (hunk 2 above)
Comment (cylon) [reply public/sass/components/_tooltip.scss]: OK, I'll put it back and do some testing.  When I had left it there it didn't seem like the dynamic changes were taking effect

Comment (cylon) [reply public/sass/components/_tooltip.scss]: I'm not sure why it wasn't taking effect before, but after putting it back the per element dynamic changes are working.

**Comment by torkelo** [comment]
ok, so the fix looks ok. But wondering if we really need this. Only one report, and I just tried to replicate the original report and could not get the tooltip on the graph below to be placed like that.

Tried many different combos of sized graph below and above with a different number of series. 
  
image

**Comment by torkelo** [comment]
First thanks for contributing and wanting to help make Grafana better! 

I am sadly closing this, we are working on a new react based graph panel & tooltip. And we prefer not to complicate the existing implementations to solve what seems like not non-critical bugs. So will hold of on fixing this bug for now.

**Pull request closed without merging**